Rename metric instruments to match feature-freeze API specification (#2202)
open-telemetry/opentelemetry-go: CHANGELOG.md

### Changed

> - Metric instruments have been renamed to match the (stable) metric API speciication:

Reviewer: [right?](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md)

```suggestion
- Metric instruments have been renamed to match the (feature-frozen) metric API speciication:
```

> - Metric instruments have been renamed to match the (feature-frozen) metric API speciication:

Author: ```suggestion
- Metric instruments have been renamed to match the (feature-frozen) metric API speciication:
```
```suggestion
- Metric instruments have been renamed to match the (feature-frozen) metric API specification:
```

  - ValueRecorder becomes Histogram
  - ValueObserver becomes Gauge
  - SumObserver becomes CounterObserver
  - UpDownSumObserver becomes UpDownCounterObserver

>   The API is still considered experimental. (#2202)

Reviewer: ```suggestion
  The API exported from this project is still considered experimental. (#2202)
```
